src/pulsar_client_manager.erl

```diff
@@ -0,0 +1,381 @@
+%% Copyright (c) 2013-2019 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+
+-module(pulsar_client_manager).
+
+-feature(maybe_expr, enable).
+
+-behaviour(gen_server).
+
+-export([start_link/3]).
+
+%% gen_server Callbacks
+-export([ init/1
+        , handle_call/3
+        , handle_cast/2
+        , handle_info/2
+        , terminate/2
+        ]).
+
+-export([ get_topic_metadata/3
+        , lookup_topic/3
+        , lookup_topic_async/2
+        ]).
+
+-export([ get_status/2
+        , get_alive_pulsar_url/2
+        , get_workers/2
```

Comment:
this is only used in tests ? maybe move to a separate `-export([...]).` block

Reply:
can be useful for introspection as well